coordinator/app/src/test/kotlin/net/consensys/zkevm/coordinator/blockcreation/BlockCreationMonitorTest.kt

```diff
@@ -111,427 +114,259 @@ class BlockCreationMonitorTest {
-        .untilAsserted {
-          verify(blockCreationListener, times(1)).acceptBlock(BlockCreated(payload))
-          assertThat(monitor.nexBlockNumberToFetch).isEqualTo(startingBlockNumberInclusive + 1)
-        }
-      testContext.completeNow()
-    }
-      .whenException(testContext::failNow)
+  fun `should be resilient to connection failures`() {
+    monitor = createBlockCreationMonitor(
+      startingBlockNumberExclusive = 99
+    )
+
+    setupFakeExecutionLayerWithBlocks(createBlocks(startBlockNumber = 99u, numberOfBlocks = 200))
+
+    monitor.start()
+    await()
+      .atMost(20.seconds.toJavaDuration())
+      .untilAsserted {
+        assertThat(blockCreationListener.blocksReceived).isNotEmpty
+        assertThat(blockCreationListener.blocksReceived.last().number).isGreaterThanOrEqualTo(103u)
+      }
+    fakeL2RpcNode.stopHttpServer()
+    val lastBlockReceived = blockCreationListener.blocksReceived.last().number
+
+    // Wait for a while to make sure no more blocks are fetched
+    await().atLeast(config.pollingInterval.times(2).toJavaDuration())
+    fakeL2RpcNode.resumeHttpServer()
+    await()
+      .atMost(20.seconds.toJavaDuration())
+      .untilAsserted {
+        assertThat(blockCreationListener.blocksReceived).isNotEmpty
+        assertThat(blockCreationListener.blocksReceived.last().number).isGreaterThan(lastBlockReceived)
+      }
   }
 
   @Test
-  fun `should stop when reorg is detected above blocksToFinalization limit - manual intervention necessary`(
-    testContext: VertxTestContext
-  ) {
-    val payload =
-      executionPayloadV1(blockNumber = startingBlockNumberInclusive, parentHash = parentHash)
-    val payload2 =
-      executionPayloadV1(
-        blockNumber = startingBlockNumberInclusive + 1,
-        parentHash = ByteArrayExt.random32()
-      )
-    val headBlockNumber = startingBlockNumberInclusive + config.blocksToFinalization
-    whenever(web3jClient.ethBlockNumber())
-      .thenReturn(SafeFuture.completedFuture(BigInteger.valueOf(headBlockNumber)))
-      .thenReturn(SafeFuture.completedFuture(BigInteger.valueOf(headBlockNumber + 1)))
-    whenever(web3jClient.ethGetExecutionPayloadByNumber(any()))
-      .thenReturn(SafeFuture.completedFuture(payload))
-      .thenReturn(SafeFuture.completedFuture(payload2))
-    whenever(blockCreationListener.acceptBlock(any())).thenReturn(SafeFuture.completedFuture(Unit))
-
-    monitor.start().thenApply {
-      await()
-        .timeout(1.seconds.toJavaDuration())
-        .untilAsserted {
-          verify(blockCreationListener, times(1)).acceptBlock(BlockCreated(payload))
-          verify(blockCreationListener, never()).acceptBlock(BlockCreated(payload2))
-          assertThat(monitor.nexBlockNumberToFetch).isEqualTo(startingBlockNumberInclusive + 1)
-        }
-      testContext.completeNow()
+  fun `should stop when reorg is detected above blocksToFinalization limit - manual intervention necessary`() {
```

Comment:
Maybe we can instead remove reorg logic completely?

Reply:
I thought about that, It would be less work for me. However, since it's already coded and I don't have 100% assurance it won't happen I left it and re-wrote the tests. I suggest when we have Beta V2 with new CL and instant finality.

Reply:
I see it more this way:
- If it was "let's implement this safe guard now in case a reorg happens", I would not advocate for it because as you mentioned it seems pretty stable
- But now, effort was already spent, the code is super isolated in a single class and fairly easy to understand, and if it happens it can be very nasty to fix... Also, are changing our consensus mechanism which until it gets battled tested this can be useful IMO. So I feel more comfortable leaving for now it's hurting..